components/KeyboardAvoidingLayout.js

```diff
@@ -0,0 +1,45 @@
+const T = require('prop-types');
+const { KeyboardAvoidingView, Platform, ScrollView } = require('react-native');
+const { useHeaderHeight } = require('@react-navigation/stack');
+
+const internals = {};
+
+exports.KeyboardAvoidingLayout = function KeyboardAvoidingLayout({ children }) {
+
+    const headerHeight = useHeaderHeight();
+
+    return (
+        <KeyboardAvoidingView
+            behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+            style={{ flex: 1 }}
+            keyboardVerticalOffset={headerHeight}
```

Comment:
It's apparently a quirk of `KeyboardAvoidingView` that the header height isn't accounted for in offset calculations. The view works perfectly on views where `headerShown: false`, but in other views where a header is present, content is still clipped _just a little_ (`64px` or so). There's a `react-native issue` (`#13393`) that was opened in April 2017 about this (I don't want to publicly link because I'm self-conscious about how GitHub handles mentions in public repositories), but it was closed in December 2019.

The consensus seems to be that this component can feel punishing to work with, especially at first, so it's not just us.